service-config/src/main/java/com/palantir/config/service/ServiceDiscoveryConfiguration.java

```diff
@@ -0,0 +1,110 @@
+     * {@link ServiceConfiguration}.
+     */
+    @JsonProperty("apiToken")
+    public abstract Optional<BearerToken> getDefaultApiToken();
+
+    /**
+     * Fallback SSL Configuration to be used if the service specific SSL configuration is not defined in the
+     * {@link ServiceConfiguration}.
+     */
+    @JsonProperty("security")
+    public abstract Optional<SslConfiguration> getDefaultSecurity();
+
+    @JsonProperty("services")
+    abstract Map<String, ServiceConfiguration> originalServices();
+
+    /**
+     * A map of {@code serviceName}s as the keys and their respective {@link ServiceConfiguration}s as the values.
+     */
+    @Lazy
+    @SuppressWarnings("checkstyle:designforextension")
+    public Map<String, ServiceConfiguration> getServices() {
+        Map<String, ServiceConfiguration> intializedServices = new HashMap<String, ServiceConfiguration>();
+
+        for (Map.Entry<String, ServiceConfiguration> entry : originalServices().entrySet()) {
+            ServiceConfiguration serviceConfig = entry.getValue();
+            Optional<BearerToken> apiToken = serviceConfig.apiToken().or(getDefaultApiToken());
+            Optional<SslConfiguration> security = serviceConfig.security().or(getDefaultSecurity());
+            intializedServices.put(entry.getKey(),
+                    ImmutableServiceConfiguration.of(apiToken, security, serviceConfig.uris()));
+        }
+
+        return Collections.unmodifiableMap(intializedServices);
+    }
+
+    /**
+     * Uses the provided service name as the key to retrieve the API token for the respective
+     * {@link ServiceConfiguration}. If the API token is not defined in the {@link ServiceConfiguration}, the default
+     * API token will be returned.
+     *
+     * @param serviceName
+     *        the name of the service to be used to retrieve the respective {@link ServiceConfiguration}
+     * @return the API token for a service based on the input service name
+     */
+    public final Optional<BearerToken> getApiToken(String serviceName) {
+        ServiceConfiguration serviceConfig =
+                Preconditions.checkNotNull(
+                        getServices().get(serviceName), "Unable to find the configuration for " + serviceName + ".");
+
+        return serviceConfig.apiToken();
+    }
+
+    /**
+     * Uses the service name as the key to retrieve the {@link SslConfiguration} for the respective
+     * {@link ServiceConfiguration}. If the {@link SslConfiguration} is not defined in the {@link ServiceConfiguration},
+     * the default {@link SslConfiguration} will be returned.
+     *
+     * @param serviceName
+     *        the name of the service to be used to retrieve the respective {@link ServiceConfiguration}
+     * @return the {@link ServiceConfiguration} for a service based on the input service name
+     */
+    public final Optional<SslConfiguration> getSslConfiguration(String serviceName) {
```

Comment:
not really, the original impl has logic to use default apiToken and security if the service specific ones aren't present. i am fine with adding that for uris.